src/build_options.rs

```diff
@@ -725,6 +691,166 @@ impl BuildOptions {
     }
 }
 
+/// Checks for bridge/platform type edge cases
+fn validate_bridge_type(
+    bridge: &BridgeModel,
+    target: &Target,
+    platform_tags: &[PlatformTag],
+) -> Result<()> {
+    match bridge {
+        BridgeModel::Bin(None) => {
+            // Only support two different kind of platform tags when compiling to musl target without any binding crates
+            if platform_tags.iter().any(|tag| tag.is_musllinux()) && !target.is_musl_target() {
+                bail!(
+                    "Cannot mix musllinux and manylinux platform tags when compiling to {}",
+                    target.target_triple()
+                );
+            }
+
+            #[allow(clippy::comparison_chain)]
+            if platform_tags.len() > 2 {
+                bail!(
+                    "Expected only one or two platform tags but found {}",
+                    platform_tags.len()
+                );
+            } else if platform_tags.len() == 2 {
+                // The two platform tags can't be the same kind
+                let tag_types = platform_tags
+                    .iter()
+                    .map(|tag| tag.is_musllinux())
+                    .collect::<HashSet<_>>();
+                if tag_types.len() == 1 {
+                    bail!(
+                        "Expected only one platform tag but found {}",
+                        platform_tags.len()
+                    );
+                }
+            }
+        }
+        _ => {
+            if platform_tags.len() > 1 {
+                bail!(
+                    "Expected only one platform tag but found {}",
+                    platform_tags.len()
+                );
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Returns the merged list of dependencies
+///
+/// Ideally, we'd like to parse the version constraints from the user and merge them with those
+/// from the transitive requirements, but that's not implemented yet
+fn collect_transitive_requires(
+    main_dependencies: &[String],
+    metadata: &Metadata,
+) -> Result<Vec<String>> {
+    let mut merged_requires = main_dependencies.to_vec();
```

Comment:
Since you have ownership of `metadata21`, why not pass `main_dependencies` by value, e.g. `take(&mut metadata32.requires_dist)`?